### PR TITLE
[Filesystem] Check that directory is writable after created it in dumpFile()

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/20612#discussion_r95077788
| License       | MIT
| Doc PR        | n/a 

In case permissions have been changed meanwhile

### CODE REVIEW

File: src/Symfony/Component/Filesystem/Filesystem.php [hunk 1]
```diff
@@ -498,7 +498,9 @@ public function dumpFile($filename, $content, $mode = 0666)
 
         if (!is_dir($dir)) {
             $this->mkdir($dir);
-        } elseif (!is_writable($dir)) {
+        }
+
+        if (!is_writable($dir)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Filesystem/Filesystem.php]: this check is already bundled into file_put_contents' failure handling. You're just making this case more explicit, but file_put_contents can fail for more reasons.
So, should it be considered a new feature? Should we use error_get_last to generate a more useful error message? should this go on master?

Comment (xabbuh) [inline src/Symfony/Component/Filesystem/Filesystem.php]: The handling was already there before. It just didn't take into account that the directory could have been created by another process.

Comment (nicolas-grekas) [inline src/Symfony/Component/Filesystem/Filesystem.php]: got it :)

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by xabbuh** [comment]
👍 

Status: Reviewed

**Comment by javiereguiluz** [comment]
👍

**Comment by fabpot** [comment]
Thank you @chalasr.

**Pull request merged**